sdk/src/Services/EC2/Custom/AmazonEC2Client.Extensions.cs

```diff
@@ -112,7 +112,7 @@ private static Dictionary<Type, DryRunInfo> MethodCache
                     _methodCache = new Dictionary<Type, DryRunInfo>();
 
                     var ec2RequestType = typeof(AmazonEC2Request);
-                    var allMembers = typeof(AmazonEC2Client).GetMembers();
+                    var allMembers = typeof(AmazonEC2Client).GetMembers(BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic);
```

Comment:
I don't quite understand why you had to add `BindingFlags.NonPublic`, there are no private methods added in this PR.

I wonder if this could break AOT applications (probably need @normj's input here).

Reply:
We needed internal members to access synchronous versions of methods which are marked as internal. I had discussed this with @normj in person. 😊

Reply:
@dscpinheiro @ashishdhingra I built a Native AOT console application calling dry run with and with out this change and had no trim warnings and the app ran successful. I believe since the `Type` can be known at compile the Native AOT compiler knows not to trim the methods out.